website/public/js/controllers/tasksCtrl.js

```diff
@@ -33,11 +36,16 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
       var newTask = {
         text: task,
         type: listDef.type,
-        tags: _.transform(User.user.filters, function(m,v,k){
-          if (v) m[k]=v;
-        })
+        tags: _.transform(task.tags, function(m, v, k) {
```

Comment:
I this should rely on User.user.filters. it's used to add the selected tags to the newly created task